setup/kubernetes/google.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Set Up a Google Kubernetes Engine Cluster
+description: Learn how to set up a GKE cluster for your Coder deployment.
+---
+
+This guide shows you how to set up a Google Kubernetes Engine (GKE) cluster to
+which Coder can deploy.
+
+## Step 1: Set Up the GKE Cluster
+
+You can set up a GKE cluster:
+
+1. Using the cloud console
+2. Using the command line
+
+### Option 1: Use the Cloud Console
+
+1. Log in to Google Cloud Console and navigate to the Kubernetes Engine page
+2. Click **CREATE CLUSTER**
+3. Provide a **Name** for your cluster and indicate the number of nodes
+4. Select the **Location type** you'd like, then choose the desired zone or
+   region for your cluster
+5. Choose a **Master** version; Coder works with versions
+   greater than 1.14.6
+6. In the left sidebar, select your **Node Pool** and click **Nodes**. Select a
+   **Machine Configuration** that will work appropriately for your expected
+   workloads; any option is fine as long as its resource allocation is greater
+   than the [resource requirements specified](../requirements.md)
+7. In the left sidebar, click **Networking**. Near the bottom, select **Enable
+   HTTP load balancing** to disable the GCE ingress controller, as well as
+   **Enable network policy** to enable the **networking.k8s.io/v1** Kubernetes
+   extension
+8. Click **Create** to proceed.
+
+### Option 2: Use the Command Line
+
+Before proceeding, make sure that the [gcloud
+CLI](https://cloud.google.com/sdk/docs/quickstarts) is installed on your
+machine.
+
+The following will spin up a Kubernetes cluster using the `gcloud` command (be
+sure to replace the parameters and environment variables as needed to reflect
+the needs of your environment):
+
+```bash
+PROJECT_ID="MY_PROJECT_ID" CLUSTER_NAME="MY_CLUSTER_NAME" \
+  gcloud beta container --project "$PROJECT_ID" \
```

Comment:
This command is a bit outdated. It doesn't support CVMs.

We should try to minimize the amount of flags too for customer sanity. What do you think about only documenting the CLI route? Customers can infer from the CLI options to adjust via the UI.

Reply:
I think documenting the CLI route only is fine